remove unused legacy mixer store code

jmuk opened the pull request:
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
none
```

jmuk commented:
/assign @guptasu 
/assign @geeknoid

geeknoid commented on mixer/pkg/server/server.go:
```diff
@@ -253,3 +253,13 @@ func (s *Server) Close() error {
 	return s.listener.Addr()
 }
+
+// InventoryMap converts adapter inventory to a builder map.
```
Is there a better place for this? This function doesn't fit the overall functionality that is offered by the server package.

guptasu commented:
I think we had a bug earlier. The server.go code should use the function `config.AdapterInfoMap` instead of `InventoryMap`. AdapterInfoMap does validation as well and make sure there are no name collisions. I think we should just delete the `InventoryMap` function from `adapterInfoRegistry.go` and may be leave everything as is. WDYT ?

geeknoid commented:
For example, seems like this function could go into the adapter package.
This package exposes the Inventory() function already, so it could also
expose InventoryMap...

On Fri, Dec 15, 2017 at 2:48 PM, codecov[bot] <notifications@github.com>
wrote:

> Codecov <https://codecov.io/gh/istio/istio/pull/2207?src=pr&el=h1> Report
>
> Merging #2207 <https://codecov.io/gh/istio/istio/pull/2207?src=pr&el=desc>
> into master
> <https://codecov.io/gh/istio/istio/commit/6706c9c9aeed3bb0a575382bc89f33a2d9c9ce60?src=pr&el=desc>
> will *increase* coverage by 0.14%.
> The diff coverage is 0%.
>
> [image: Impacted file tree graph]
> <https://codecov.io/gh/istio/istio/pull/2207?src=pr&el=tree>
>
> @@            Coverage Diff             @@##           master    #2207      +/-   ##
> ==========================================+ Coverage   78.04%   78.18%   +0.14%
> ==========================================
>   Files         209      201       -8
>   Lines       18473    18152     -321
> ==========================================- Hits        14417    14192     -225     + Misses       3297     3216      -81     + Partials      759      744      -15
>
> Flag Coverage Δ
> #broker 47.27% <ø> (ø) ⬆️
> #mixer 78.03% <0%> (+0.26%) ⬆️
> #pilot 79.09% <ø> (-0.05%) ⬇️
> #security 87.24% <ø> (ø) ⬆️
> Impacted Files
> <https://codecov.io/gh/istio/istio/pull/2207?src=pr&el=tree> Coverage Δ
> mixer/pkg/config/store/store2.go
> <https://codecov.io/gh/istio/istio/pull/2207/diff?src=pr&el=tree#diff-bWl4ZXIvcGtnL2NvbmZpZy9zdG9yZS9zdG9yZTIuZ28=> 94.28%
> <ø> (ø) ⬆️
> mixer/cmd/mixs/cmd/validator.go
> <https://codecov.io/gh/istio/istio/pull/2207/diff?src=pr&el=tree#diff-bWl4ZXIvY21kL21peHMvY21kL3ZhbGlkYXRvci5nbw==> 0%
> <0%> (ø) ⬆️
> pilot/platform/eureka/controller.go
> <https://codecov.io/gh/istio/istio/pull/2207/diff?src=pr&el=tree#diff-cGlsb3QvcGxhdGZvcm0vZXVyZWthL2NvbnRyb2xsZXIuZ28=> 81.25%
> <0%> (-9.38%) ⬇️
> mixer/adapter/stackdriver/metric/bufferedClient.go
> <https://codecov.io/gh/istio/istio/pull/2207/diff?src=pr&el=tree#diff-bWl4ZXIvYWRhcHRlci9zdGFja2RyaXZlci9tZXRyaWMvYnVmZmVyZWRDbGllbnQuZ28=> 90.32%
> <0%> (-3.23%) ⬇️
> mixer/pkg/config/inventory.go
> <https://codecov.io/gh/istio/istio/pull/2207/diff?src=pr&el=tree#diff-bWl4ZXIvcGtnL2NvbmZpZy9pbnZlbnRvcnkuZ28=>
> mixer/pkg/server/args.go
> <https://codecov.io/gh/istio/istio/pull/2207/diff?src=pr&el=tree#diff-bWl4ZXIvcGtnL3NlcnZlci9hcmdzLmdv>
> mixer/pkg/server/monitoring.go
> <https://codecov.io/gh/istio/istio/pull/2207/diff?src=pr&el=tree#diff-bWl4ZXIvcGtnL3NlcnZlci9tb25pdG9yaW5nLmdv>
> mixer/pkg/server/tracing.go
> <https://codecov.io/gh/istio/istio/pull/2207/diff?src=pr&el=tree#diff-bWl4ZXIvcGtnL3NlcnZlci90cmFjaW5nLmdv>
> ------------------------------
>
> Continue to review full report at Codecov
> <https://codecov.io/gh/istio/istio/pull/2207?src=pr&el=continue>.
>
> *Legend* - Click here to learn more
> <https://docs.codecov.io/docs/codecov-delta>
> Δ = absolute <relative> (impact), ø = not affected, ? = missing data
> Powered by Codecov
> <https://codecov.io/gh/istio/istio/pull/2207?src=pr&el=footer>. Last
> update 6706c9c...17d6cd0
> <https://codecov.io/gh/istio/istio/pull/2207?src=pr&el=lastupdated>. Read
> the comment docs <https://docs.codecov.io/docs/pull-request-comments>.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/2207#issuecomment-352131128>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AVucHVDNIWqHy4_7mq0J3KOR0DkfNT21ks5tAvc3gaJpZM4REFIx>
> .
>

jmuk commented:
Hmm, okay, so:
- switched to use `AdapterInfoMap`. I still think it's better to do validation like this on codegen phase, but that's another topic.
- finding the place for adapterInfoRegistry.go is a bit hard, it can't be in `mixer/pkg/adapter` due to some cycle dependency on test. That would be another topic, maybe it would be `mixer/pkg/adapter_registry`, but let's leave it as-is for this specific PR.

Updated the PR. Please take another look.

guptasu commented:
/lgtm
